packages/cli/src/ui/components/shared/text-buffer.ts

```diff
@@ -526,148 +527,110 @@ export function useTextBuffer({
     return _restoreState(state);
   }, [redoStack, lines, cursorRow, cursorCol, _restoreState]);
 
-  const insertStr = useCallback(
-    (str: string): boolean => {
-      dbg('insertStr', { str, beforeCursor: [cursorRow, cursorCol] });
-      if (str === '') return false;
+  const applyOperations = useCallback((ops: UpdateOperation[]) => {
+    if (ops.length === 0) return;
+    setOpQueue((prev) => [...prev, ...ops]);
```

Comment:
this is probably safe because you are adding prev but still I think switching to the proper idioms is likely safer.